fxgl-trade/src/main/kotlin/com/almasb/fxgl/inventory/Inventory.kt

```diff
@@ -26,70 +27,148 @@ class Inventory<T>(var capacity: Int) {
     private val log = Logger.get(javaClass)
 
     private val items = FXCollections.observableArrayList<T>()
+
     private val itemsProperty = FXCollections.unmodifiableObservableList(items)
 
+    private val stacks = FXCollections.observableArrayList<StackData<T>>()
+
+    private val stacksProperty = FXCollections.unmodifiableObservableList(stacks)
+
     private val itemsData = hashMapOf<T, ItemData<T>>()
 
+    private val stacksData = hashMapOf<Int, StackData<T>>()
```

Comment:
I currently don't see why we we need an `Int -> Data` mapping. I thought we could simply add a `Map<T, List<ItemData<T> >>`. I haven't given this much thought, so I don't know if this works or not, but can you consider this option?

If the above works, then we don't really need `Stack` and any of its related code, right? Then 

`fun incrementQuantity(item: T, amount: Int): Boolean` is just going to:

```
if (item in map) {
    var list = map.get(item)
   // find first item data in list that can hold more qty or add new ItemData
} else {
    add new ItemData
}
```

I will review the rest once we have resolved this.

Reply:
The `Int -> Data` mapping was unnecessary but I thought it might help when creating a grid-style view if each stack had an ID representing it's current slot in the inventory.  I created the StackData class to represent individual stacks and to not have repeat attributes from ItemData (each stack of same item will have the same description, stackMax, name, etc..). Additionally I was using the StackData type to select specific stacks in the InventoryView  with:

```private val stacks = FXCollections.observableArrayList<StackData<T>>()```

Either way I am starting from scratch with your comments in mind and I'm not sure if storing maxStack in ItemData is possible.  If `isRemoveItemIfQty0 == false` and maximum stack is an attribute on the `ItemData` this may result in `T -> empty ObservableList` with no reference to what the stackMax was. 
   - I can create a `map( T -> Int)` where Int is stackMax for each item, T.
   - I can create a make itemsData values int a pair to track stackMax:
       ```private val itemsData = hashMapOf<T, Pair<Int, ItemData<T>>>()``` 



Or I can simplify my original commit without IDs for each stack.

Let me know if I'm way off here, thanks again for the input.

Reply:
I see, thanks for the explanation. 

I can see now how `Int -> Data` would be useful for that, but there is currently no use case for this. In the long term it would be good to have something like `StackData`. I appreciate the future-proof solution here, but incremental changes are easier to manage.  So I'd propose to tweak the current PR to minimise changes, whilst introducing this new feature.

> If isRemoveItemIfQty0 == false and maximum stack is an attribute on the ItemData this may result in T -> empty ObservableList with no reference to what the stackMax was.

Good point. Can we decompose the problem to one single (universal) max quantity variable. This will simplify the current PR. We can then discuss how we can handle max quantity on a per-item basis.

In the next (separate) PR we can consider how to clean up the implementation, including "repeat attributes from ItemData", for now `List<ItemData>` should be sufficient I believe.